fedimint-api-client/src/lib.rs

```diff
@@ -23,65 +24,74 @@ pub mod api;
 /// Client query system
 pub mod query;
 
-/// Tries to download the client config from the federation,
-/// attempts to retry teb times before giving up.
-pub async fn download_from_invite_code(invite_code: &InviteCode) -> anyhow::Result<ClientConfig> {
-    debug!("Downloading client config from {:?}", invite_code);
+impl Connector {
+    /// Tries to download the [`ClientConfig`] from the federation with an
+    /// specified [`Connector`] variant, attempts to retry ten times before
+    /// giving up.
+    pub async fn download_from_invite_code(
+        &self,
+        invite_code: &InviteCode,
+    ) -> anyhow::Result<ClientConfig> {
```

Comment:
maybe this is fine as free function

```rust
pub async fn download_from_invite_code(invite_code: &InviteCode, connector: &Connector)
```

Reply:
I did this way to reduce the number of fns that needs to keep passing the `connector` parameter around.

I'm fine with reverting it to a free function though.